FBTweak/_FBTweakCollectionViewController.m

```diff
@@ -28,16 +28,33 @@ - (instancetype)initWithTweakCategory:(FBTweakCategory *)category
     _tweakCategory = category;
     self.clearsSelectionOnViewWillAppear = YES;
     self.title = _tweakCategory.name;
+
+    [(id)self.tweakCategory addObserver:self
+                             forKeyPath:NSStringFromSelector(@selector(tweakCollections))
+                                options:NSKeyValueObservingOptionInitial | NSKeyValueObservingOptionNew
+                                context:nil];
+
     self.refreshControl = [[UIRefreshControl alloc] init];
-    [self.refreshControl addTarget:self
-                            action:@selector(_reloadData)
+    [self.refreshControl addTarget:self action:@selector(_updateCategory)
                   forControlEvents:UIControlEventValueChanged];
-    [self _reloadData];
   }
   
   return self;
 }
 
+- (void)dealloc
+{
+  [(id)self.tweakCategory removeObserver:self
+                              forKeyPath:NSStringFromSelector(@selector(tweakCollections))];
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(NSArray<FBTweakCollection *> *)tweakCollections change:(NSDictionary *)change context:(void *)context
+{
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Enclose this block in the following construct for safety:
```objc
  if ([keyPath isEqualToString:NSStringFromSelector(@selector(tweakCollections))]) {
    // the dispatch_async block goes here.
  }
```

Reply:
Used this to do bail-fast.